/*++

    Бросаем число.

    Плюсовики привыкли, что исключения имеют какую-то свою иерархию и
    каждый класс имеет свое конкретное назначение в контексте отображения
    ошибки.

    А что если бросить, что-то несвязанное с исключением?
    Например, какой-нибуть тривиальный тип int. Это вообще возможно?
    Как оказалось в С++ можно бросать все, что угодно, кроме объектов
    неполных типов, абстрактных классов и указателей на неполный тип.
    Даже указатель void можно.

    Как и число...

--*/

#include <iostream>
#include <cstdlib>
#include <cstdint>

auto foo() -> void
{ throw 42; }

int main()
{
    using std::cout;
    using std::endl;

    try { ::foo(); }
    catch (int32_t num) {
        cout
            << '\n'
            << "E: "
            << num
            << endl;
    }

    std::exit(0);
}
